modules/eth/lighthouse-beacon.nix

```diff
@@ -2,6 +2,13 @@
 with lib;
 let
   cfg = config.lighthouse;
+  # split endpoint to address and port
+  endpointRegex = "(https?://)?([^:/]+):([0-9]+)?";
```

Comment:
I have to think about this a bit. I like the fact that we can include the port in the lighthouse config, but I think it would be more useful to have the port as a distinct variable in Nix to be able to compose to other commands as well. Also, I have a slight worry about regexes, and the fact that this seems to bind into `https`, which we don't currently use. To say, I don't think the nodes will run properly if this change is added, since we don't have local HTTPS certificates to do authentication currently.

Is it possible to run some arbitrary command in place of here to do the splitting? So, could I do the splitting by running some command that is available from `pkgs` instead?

Reply:
More about the regex: users would be able to input URLs like `http://192.168.1.1:1337`, `https://192.168.1.1:1337` or `192.168.1.1:1337`. Regardless of the input format, the resulting list would always have the IP address as the second element and the port number as the third. I don't see a problem with this if the port number does not have to be evaluated on it's own. 

```
nix-repl> endpointRegex = "(https?://)?([^:/]+):([0-9]+)?"

nix-repl> builtins.match endpointRegex "http://192.168.1.1:1337"
[ "http://" "192.168.1.1" "1337" ]

nix-repl> builtins.match endpointRegex "https://192.168.1.1:1337"
[ "https://" "192.168.1.1" "1337" ]

nix-repl> builtins.match endpointRegex "192.168.1.1:1337"
[ null "192.168.1.1" "1337" ]
```
Trailing slash or a path would be a problem at the moment but can be taken into account like so:
```
nix-repl> endpointRegex = "(https?://)?([^:/]+):([0-9]+)(/[^/]*)?"

nix-repl> builtins.match endpointRegex "192.168.1.1:1337/eth"
[ null "192.168.1.1" "1337" "/eth" ]

nix-repl> builtins.match endpointRegex "http://192.168.1.1:1337/"
[ "http://" "192.168.1.1" "1337" "/" ]

nix-repl> builtins.match endpointRegex "https://192.168.1.1:1337"
[ "https://" "192.168.1.1" "1337" null ]
```

Reply:
> endpointRegex = "(https?://)?([^:/]+):([0-9]+)?"

this one seems good. i did this but realized we probably cant use it anyway since bqn would be needed to be built into nix: https://bqnpad.mechanize.systems/s?bqn=eyJkb2MiOiJQcmUg4oaQIOKGleKImOKJoOKJoeKNi1xuUmVzZXRBdCDihpAg4oqSJzon4o234oqiXG4owqxQcmUgwqgoMeKGk%2BKGkSBSZXNldEF0IFwiMTkyLjE2OC4xMDAuMTA6NzAwMFwiKSkg4oq4IOKKlCBcIjE5Mi4xNjguMTAwLjEwOjcwMDBcIlxuIiwicHJldlNlc3Npb25zIjpbXSwiY3VycmVudFNlc3Npb24iOnsiY2VsbHMiOltdLCJjcmVhdGVkQXQiOjE2ODQxMDkyMjU1OTR9LCJjdXJyZW50Q2VsbCI6eyJmcm9tIjowLCJ0byI6OTksInJlc3VsdCI6bnVsbH19